compiler-rt/lib/radsan/radsan.cpp

```diff
@@ -7,26 +7,137 @@
 */
 
 #include <radsan/radsan.h>
+
 #include <radsan/radsan_context.h>
+#include <radsan/radsan_flags.h>
 #include <radsan/radsan_interceptors.h>
-#include <unistd.h>
+
+#include "sanitizer_common/sanitizer_atomic.h"
+#include "sanitizer_common/sanitizer_common.h"
+#include "sanitizer_common/sanitizer_flag_parser.h"
+#include "sanitizer_common/sanitizer_flags.h"
+
+using namespace __sanitizer;
+
+namespace radsan {
+static __sanitizer::atomic_uint8_t radsan_inited{};
+static __sanitizer::atomic_uint8_t radsan_init_is_running{};
```

Comment:
Another naive question: I can't picture a situation in my head where a second thread could be created during `radsan_init`, because I thought it's meant to be called before `main` - have you seen this happen elsewhere? Unless maybe this protects against the situation where a user might have added their own thread-creating functions into the `.preinit` array as well?

Reply:
See comment on .preinit below!